### PR TITLE
leiningen: update to 2.10.0.

### PR DESCRIPTION
**Pull request by AdamFrey**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

### CODE REVIEW

File: srcpkgs/leiningen/template [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
-distfiles="https://github.com/technomancy/${pkgname}/archive/${version}.tar.gz
- https://github.com/technomancy/${pkgname}/releases/download/${version}/leiningen-${version}-standalone.zip"
-checksum="a29b45966e5cc1a37d5dc07fe436ed7cb172c88c53d44a049956ff53a096d43e
- df490c98bfe8d667bc5d83b80238528877234c285d0d48f61a4c8743c2db1eea"
-skip_extraction="leiningen-${version}-standalone.zip"
+distfiles="https://codeberg.org/${pkgname}/${pkgname}/archive/${version}.tar.gz
+ https://codeberg.org/attachments/${jar_id}"
+checksum="6b04dbe0934ab4c34dc27b0055ad290d2ba9c0f454c1bda7defa58d62946b743
+ d27299bad34075ac2864d0bd0559f835c6e2c476c0b0a283bcbdb574fdadbb34"
+_jar_id=43cebda5-a7c2-405b-b641-5143a00051b5
```
Comment (paper42) [inline srcpkgs/leiningen/template:14]: put jar_id after revision.

File: srcpkgs/leiningen/template [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+
+do_extract() {
+	bsdtar xf ${XBPS_SRCDISTDIR}/${pkgname}-${version}/${version}.tar.gz
+}
```
Comment (paper42) [inline srcpkgs/leiningen/template:18]: why is this required? skip_extraction is not used

Comment (AdamFrey) [reply srcpkgs/leiningen/template]: Oops, I was confused about the `skip_extraction` functionality. I force-pushed [a new commit](https://github.com/void-linux/void-packages/pull/41825/commits/aac814bfdaa78cf1a4585b4975a1c9cc4573aa1f) which avoids `do_extract`.

File: srcpkgs/leiningen/template (hunk 1 above)
Comment (AdamFrey) [reply srcpkgs/leiningen/template]: :+1: https://github.com/void-linux/void-packages/pull/41825/commits/aac814bfdaa78cf1a4585b4975a1c9cc4573aa1f

File: srcpkgs/leiningen/template [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
-version=2.9.5
+version=2.10.0
 revision=1
+_jar_id=43cebda5-a7c2-405b-b641-5143a00051b5
 depends="bash virtual?java-runtime"
 short_desc="Automate Clojure projects without setting your hair on fire"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Adam Frey <adam@adamfrey.me>"
 license="EPL-2.0"
 homepage="http://leiningen.org/"
```
Comment (icp1994) [inline srcpkgs/leiningen/template:10]:
```suggestion
license="EPL-1.0"
homepage="https://leiningen.org/"
changelog="https://codeberg.org/leiningen/leiningen/raw/branch/main/NEWS.md"
```

**Review by icp1994 — CHANGES_REQUESTED** [review]

**Comment by AdamFrey** [comment]
@icp1994 I committed your change, which seemed to cause the linter to fail, but I can't figure out why.

**Comment by paper42** [comment]
This PR now contains two commits, one with a commit message that's wrong, so it complains about that. Squash the commits to one.

**Comment by classabbyamp** [comment]
> b5aac7eac: subject does not follow CONTRIBUTING.md guildelines

see https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes

**Comment by AdamFrey** [comment]
> > [b5aac7e](https://github.com/void-linux/void-packages/commit/b5aac7eacfea7ba4cef2108fc497a44addcde857): subject does not follow CONTRIBUTING.md guildelines
> 
> see https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes

I see, thank you. I squashed my commits together and force pushed to this branch.

File: srcpkgs/leiningen/template [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
-distfiles="https://github.com/technomancy/${pkgname}/archive/${version}.tar.gz
- https://github.com/technomancy/${pkgname}/releases/download/${version}/leiningen-${version}-standalone.zip"
-checksum="a29b45966e5cc1a37d5dc07fe436ed7cb172c88c53d44a049956ff53a096d43e
- df490c98bfe8d667bc5d83b80238528877234c285d0d48f61a4c8743c2db1eea"
-skip_extraction="leiningen-${version}-standalone.zip"
+maintainer="Adam Frey <adam@adamfrey.me>"
+license="EPL-1.0"
+homepage="https://leiningen.org/"
+changelog="https://codeberg.org/leiningen/leiningen/raw/branch/main/NEWS.md"
+distfiles="https://codeberg.org/${pkgname}/${pkgname}/archive/${version}.tar.gz
```
Comment (classabbyamp) [inline srcpkgs/leiningen/template:12]:
```suggestion
distfiles="https://codeberg.org/leiningen/leiningen/archive/${version}.tar.gz
```
